### PR TITLE
Localization for Blazor WASM

### LINKED ISSUE #17513: Localization for Blazor WASM
* By default, .NET WebAssembly sets the culture using the user's language preference. See https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/languages
* Application developers can optionally configure the application culture by setting `CultureInfo.DefaultThreadCurrentCulture` \ `CultureInfo.DefaultThreadCurrentUICulture` as part of their `Program.MainAsync`.
* To minimize network costs, only satellite assemblies for the selected culture are downloaded.
* Sample app with culture picker: https://github.com/pranavkm/LocSample

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 15e7c51d-1839-1849-813a-23c4d21c64ae
* Version Independent ID: d6f07538-228e-9f96-680f-6c324caf11d6
* Content: [ASP.NET Core Blazor globalization and localization](https://docs.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-3.1)
* Content Source: [aspnetcore/blazor/globalization-localization.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/globalization-localization.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #17513 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-3.1&branch=pr-en-us-17569)

We'll be able to do more here later when there's more 🕐. For example, I think we should eventually add example code. For now 🏃😅, I recommend that we use this guidance with the linked engineering sample.

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
@pranavkm can you please review this? Thanks!

**Comment by mkArtakMSFT** [comment]
@guardrex please expect this to delay till some time next week.

File: aspnetcore/blazor/globalization-localization.md [hunk 1]
```diff
@@ -52,6 +52,18 @@ The following field types have specific formatting requirements and aren't curre
...
 
+### Blazor WebAssembly
+
+Blazor WebAssembly apps set the culture using the user's [language preference](https://developer.mozilla.org/docs/Web/API/NavigatorLanguage/languages).
+
+To explicitly configure the culture, set `CultureInfo.DefaultThreadCurrentCulture` and `CultureInfo.DefaultThreadCurrentUICulture` in `Program.Main`.
+
+By default, Blazor's linker configuration for Blazor WebAssembly apps strips out internationalization information except for locales explicitly requested. For more information and guidance on controlling the linker's behavior, see <xref:host-and-deploy/blazor/configure-linker#configure-the-linker-for-internationalization>.
+
+For a Blazor WebAssembly sample app with a culture picker, see the [LocSample](https://github.com/pranavkm/LocSample) localization sample app.
```
Comment (pranavkm) [inline aspnetcore/blazor/globalization-localization.md:63]: Should I move this to the samples repo? Maybe we want to say something to the effect of, while the culture that Blazor picks by default might be sufficient for most users, consider offering a way for users to specify their preferred locale. <Sample link here> has an example of a doing this.

Comment (guardrex) [reply aspnetcore/blazor/globalization-localization.md:63]: For today ... given the workload at this time 🏃 ... it's fine to leave it where it is. If possible, I'd like to roll this into the common sample apps *soon-ish*. The work would be done after 3.2 releases and is tracked by https://github.com/dotnet/AspNetCore.Docs/issues/14216. The goal is to minimize the number of sample apps that we need to keep up with and make releases easier to manage. If you prefer it as a standalone sample, that's ok. Let me know. :ear:

**Comment by guardrex** [comment]
@pranavkm I'm going to go ahead and merge this. Let me know if you'd prefer the LocSample hosted here instead of in your repo until the common sample apps can take on the code. If ur ok with holding it for a short while, I'll ping you when it moves into the common samples and is no longer cross-linked.

btw - I'm not aware yet that ASP.NET Core docs samples are moving to the samples repo yet ... the word hasn't been given for our samples AFAIK.

**Comment by pranavkm** [comment]
I think we merged this in a little too early. P4 hasn't shipped as yet

**Comment by guardrex** [comment]
It wasn't noted on the issue ...

https://github.com/dotnet/AspNetCore.Docs/issues/17512

I'll back it out this evening and reset the PR for P4.

**Comment by guardrex** [comment]
Will be tracked by https://github.com/dotnet/AspNetCore.Docs/issues/17808 ... I'll do it this evening and merge to live immediately.

**Pull request merged**